tests/test_dse.py

```diff
@@ -20,6 +21,42 @@
 pytestmark = skipif(['yask', 'ops'])
 
 
+@pytest.mark.parametrize('expected', [
+    ['Eq(r3, 1/h_x)',
+     'Eq(r2, 1/dt)',
+     'Eq(d[t + 1, x, y],'
+     ' 0.0833333333*r3*d[t, x - 2, y] - 0.666666667*r3*d[t, x - 1, y] +'
+     ' 0.666666667*r3*d[t, x + 1, y] - 0.0833333333*r3*d[t, x + 2, y] + 1)',
+     'Eq(a[t + 1, x, y],'
+     ' 0.0833333333*r3*a[t, x - 2, y] - 0.666666667*r3*a[t, x - 1, y] +'
+     ' 0.666666667*r3*a[t, x + 1, y] - 0.0833333333*r3*a[t, x + 2, y] + 1)',
+     'Eq(r1, -r2*a[t, x, y])',
+     'Eq(r0, r2*a[t + 1, x, y])',
+     'Eq(b[t + 1, x, y], r0 + r1 + 1)',
+     'Eq(c[t + 1, x, y], r0 + r1 + 2)']])
+def test_topological_sorting(expected):
+    """
+    Test that verifies that:
+    - Temporaries are sorted according to dependencies
+    - Temporaries are not moved before and equation that defines its rhs
```

Comment:
Maybe rephrase that? Does not seem that clear at least for me.

Reply:
I believe it's just a typo: `and` -> `an` then it makes sense :)